tests/test_adapter.py

```diff
@@ -2008,3 +2012,99 @@ def test_exception_handling(self, mock_stage, mock_download_subset,
+
+        dimensions_path = f'{self.tmp_dir}/dimensions.nc4'
+        copy('tests/data/ATL16_prefetch.nc4', dimensions_path)
+
+        all_variables_path = f'{self.tmp_dir}/variables.nc4'
+        copy('tests/data/ATL16_variables.nc4', all_variables_path)
+
+        mock_util_download.side_effect = [dmr_path, dimensions_path,
+                                          all_variables_path]
+
+        message = Message({
+            'accessToken': 'fake-token',
+                'callback': 'https://example.com/',
+                'sources': [{
+                    'collection': 'C1238589498-EEDTEST',
+                    'shortName': 'ATL16',
+                    'variables': [{'id': '',
+                                'name': self.atl16_variable,
+                                'fullPath': self.atl16_variable}]}],
+                'stagingLocation': self.staging_location,
+                'subset': {'bbox': [77, 71.5, 88, 74.75]},
+                'user': 'jlovell',
+        })
+
+        hoss = HossAdapter(message, config=config(False), catalog=self.input_stac)
+        _, output_catalog = hoss.invoke()
+
+        # Ensure that there is a single item in the output catalog with the
+        # expected asset:
+        self.assert_expected_output_catalog(output_catalog,
+                                            expected_staged_url,
+                                            expected_output_basename)
+
+        # Ensure the expected requests were made against OPeNDAP.
+        # Q JSL - Why are there two of the same calls in a few of the test_adapter.py tests?
+        self.assertEqual(mock_util_download.call_count,3)
+        mock_util_download.assert_has_calls([
+            call(f'{self.granule_url}.dmr.xml', self.tmp_dir, hoss.logger,
+                    access_token=message.accessToken, data=None, cfg=hoss.config),
+            call(f'{self.granule_url}.dap.nc4', self.tmp_dir, hoss.logger,
+                    access_token=message.accessToken, data=ANY, cfg=hoss.config),
+            call(f'{self.granule_url}.dap.nc4', self.tmp_dir, hoss.logger,
+                    access_token=message.accessToken, data=ANY, cfg=hoss.config),
+        ])
+
+        # Ensure the constraint expression for dimensions data included only
+        # dimension variables and their associated bounds variables.
+        dimensions_data = mock_util_download.call_args_list[1][1].get('data', {})
+        self.assert_valid_request_data(
+            dimensions_data,
+            {'%2Fglobal_grid_lat',
+             '%2Fglobal_grid_lon'}
+        )
+
+        # Ensure the constraint expression contains all the required variables.
+        index_range_data = mock_util_download.call_args_list[2][1].get('data', {})
+        self.assert_valid_request_data(
+            index_range_data,
+            {'%2Fglobal_asr_obs_grid%5B53%3A54%5D%5B85%3A89%5D',
+             '%2Fglobal_grid_lat%5B53%3A54%5D',
+             '%2Fglobal_grid_lon%5B85%3A89%5D'}
```

Comment:
I got these values from looking at the third call, but still had to adjust the numbers a little. I'm not entirely sure what they mean.

Reply:
what's this assertion testing? You can decode those strings to see what they mean with urllib.parse.quote/unquote

https://docs.python.org/3/library/urllib.parse.html#url-quoting

Reply:
Okay - so just to explicitly check: this bounding box is that special case, and the index ranges returned in this test are the ones you expect?

Reply:
Ha - that's weird. But cool beans. Thanks for the confirmation!